base/show.jl

```diff
@@ -2470,15 +2470,23 @@ function print_within_stacktrace(io, s...; color=:normal, bold=false)
     end
 end
 
-function show_tuple_as_call(io::IO, name::Symbol, sig::Type;
+function show_tuple_as_call(out::IO, name::Symbol, sig::Type;
                             demangle=false, kwargs=nothing, argnames=nothing,
                             qualified=false, hasfirst=true)
     # print a method signature tuple for a lambda definition
     if sig === Tuple
-        print(io, demangle ? demangle_function_name(name) : name, "(...)")
+        print(out, demangle ? demangle_function_name(name) : name, "(...)")
         return
     end
     tv = Any[]
+    if get(out, :limit, false)::Bool
+        io = IOBuffer()
+        if isa(out, IOContext)
+            io = IOContext(io, out)
+        end
+    else
+        io = out
+    end
```

Comment:
Type-unstable code like this should generally be avoided. There is no reason to not call the `IOContext` wrapper unconditionally, and it is incorrect anyways not to wrap it when making an IOBuffer that is supposed to act like the original IO (several properties still get copied over, even if it is not previously an IOContext).